src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1930,7 +1930,7 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
      * If alias length is more than 29, abbreviates column name.
      */
     protected buildColumnAlias(aliasName: string, columnName: string): string {
-        const columnAliasName = aliasName + "_" + columnName;
+        const columnAliasName = this.connection.namingStrategy.columnAliasName(aliasName, columnName);
         if (columnAliasName.length > 29 && this.connection.driver instanceof OracleDriver)
             return aliasName  + "_" + abbreviate(columnName, 2);
```

Comment:
You are not using naming strategy for abbreviate column name.

Reply:
> 1. First at all is this change required? I do not see any issue where this change will help to resolve some real life issue. Lets see what @pleerock.

We try to understand if this change is needed. The problem has occurred using an upper-snake-case nomenclature with Oracle driver. 
The current buildColumnAlias function, abbreviate for aliases with longer than 29 characters, the name of the column. In doing so though in case of similar fields ex. PRODUCT_CATEGORY_ID, PRODUCT_CATALOG_ID, you will get the same result: **PR**, and then an incorrect result of the query. The need was therefore to be able to customize this function.

> You are not using naming strategy for abbreviate column name.

Not being able to check the driver inside the new naming strategy function, the only chance that I saw was using a custom implementation into the new naming strategy function that returns a string shorter than 29 characters, to skip the condition.